### PR TITLE
`Development`: Add branch naming conventions to documentation

### PR DESCRIPTION
**Pull request by rstief**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
We wanted to add the new branch naming conventions to the documentation to help new developers see what branch names are allowed.

### Description
- Adds branch naming conventions to documentation
- Removes all outdated content from CONTRIBUTING.md
- Moves not outdated content from CONTRIBUTING.md to respective docs files.

### Steps for Testing
Visit the changed documentation:
- [Development Process](https://artemis-platform--8504.org.readthedocs.build/en/8504/dev/development-process/development-process.html#create-a-local-branch)
- [Styling](https://artemis-platform--8504.org.readthedocs.build/en/8504/dev/guidelines/client.html#styling)

View the new [CONTRIBUTING.md](https://github.com/ls1intum/Artemis/blob/6422b3cf19bd9fbc4eaf2bf6a90a7a26fe9eee72/CONTRIBUTING.md)

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: docs/dev/development-process/development-process.rst [hunk 1]
```diff
@@ -103,27 +103,40 @@ When an issue is labeled with "UI" on GitHub, UICompanion automatically attaches
...
+**Branches that do not follow this structure will get rejected by GitHub!**
+
+5. Implement the Feature
+========================
+In this step, the development team converts the detailed plans and designs outlined in the functional proposal into working code.
 This step requires careful adherence to the previously defined requirements and system architecture to ensure that the function fits seamlessly into the existing system and fulfills the specified functional and performance criteria.
 
+Naming Conventions for Branches
+--------------------------------
+
```
Comment (rstief) [inline docs/dev/development-process/development-process.rst:121]:
```suggestion
```

File: docs/dev/development-process/development-process.rst [hunk 2]
```diff
@@ -103,27 +103,37 @@ When an issue is labeled with "UI" on GitHub, UICompanion automatically attaches
 
 For more details and to get started with UICompanion, visit the `UICompanion Repository on GitHub <https://github.com/ls1intum/UICompanion>`_.
 
-4. Implement the Feature
+4. Create a local Branch
 ========================
-In this step, the development team converts the detailed plans and designs outlined in the functional proposal into working code. 
+We use the following structure for branch names: ``<type>/<area>/<short-description>``
+- Possible types are: **feature**, **chore** and **bugfix**
```
Comment (Strohgelaender) [inline docs/dev/development-process/development-process.rst:109]:
```suggestion
- Possible types are: **feature**, **chore**, and **bugfix**
```

File: docs/dev/development-process/development-process.rst [hunk 3]
```diff
@@ -103,27 +103,37 @@ When an issue is labeled with "UI" on GitHub, UICompanion automatically attaches
...
+- Possible areas are all allowed feature tags (see :ref:`pr_naming_conventions`), written kebab-case. For example, ``Programming exercises`` becomes **programming-exercises**.
+
+**Branches that do not follow this structure will get rejected by GitHub!**
```
Comment (Strohgelaender) [inline docs/dev/development-process/development-process.rst:112]:
```suggestion
**Branches that do not follow this structure will get rejected automatically!**
```

File: docs/dev/guidelines/client.rst [hunk 4]
```diff
@@ -470,3 +470,15 @@ Best Practices:
...
+===========
+
+We are using `Scss <https://sass-lang.com>`_ to write modular, reusable css. We have a couple of global scss files in ``webapp/content/scss`` but encourage `component dependent css with angular's styleUrls <https://angular.io/guide/component-styles>`_.
+
+Within the component html files, we encourage the use of `bootstrap css <https://getbootstrap.com/>`_:
+
+.. code-block:: html
+
+    <div class="d-flex ms-2">some content</div>
+
```
Comment (rstief) [inline docs/dev/guidelines/client.rst:484]:
```suggestion
18. Styling
===========

We are using `Scss <https://sass-lang.com>`_ to write modular, reusable css. We have a couple of global scss files in ``webapp/content/scss``, but encourage component dependent css using `Angular styleUrls <https://angular.io/guide/component-styles>`_.

From a methodology viewpoint we encourage the use of `BEM <http://getbem.com/introduction/>`_:

.. code-block:: scss

    .my-container {
        // container styles
        &__content {
            // content styles
            &--modifier {
                // modifier styles
            }
        }
    }

Within the component html files, we encourage the use of `bootstrap css <https://getbootstrap.com/>`_:

.. code-block:: html

    <div class="d-flex ms-2">some content</div>

```

File: docs/dev/guidelines/client.rst [hunk 5]
```diff
@@ -470,3 +470,29 @@ Best Practices:
 1. Dynamic Subscription Handling: Subscribe to topics on an as-needed basis. Unsubscribe from topics that are no longer needed to keep the number of active subscriptions within the recommended limit.
 2. Efficient Topic Aggregation: Use topic aggregation techniques to consolidate related data streams into a single subscription wherever possible. Consequently, don't create a new topic if an existing one can be reused.
 3. Small Messages: Send small messages and use DTOs. See :ref:`server-guideline-dto-usage` for more information and examples.
+
+18. Styling
```
Comment (rstief) [inline docs/dev/guidelines/client.rst:474]:
```suggestion
19. Styling
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,110 +1,6 @@
 # Contributing Guide for Artemis
 
-Please read this guide before creating a pull request, otherwise your contribution might not be approved.
+Read the [setup guide](https://docs.artemis.cit.tum.de/dev/setup.html) on how to set up your local development environment.
 
-## Branch Organization
+Before creating a pull request, please read the guidelines to the [development process](https://docs.artemis.cit.tum.de/dev/development-process.html) as well as the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines.html).
```
Comment (Jan-Thurner) [inline CONTRIBUTING.md:5]:
```suggestion
Before creating a pull request, please read the [guidelines to the development process](https://docs.artemis.cit.tum.de/dev/development-process.html) as well as the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines.html).
```

**Review by Jan-Thurner — COMMENTED** [review]
Looks good imo. I left one comment.

**Review by Jan-Thurner — APPROVED** [review]

**Review by edkaya — DISMISSED** [review]
Looks good to me 👍

**Review by laurenzfb — DISMISSED** [review]
lgtm

**Review by egekurt123 — APPROVED** [review]
LGTM

**Review by Jan-Thurner — APPROVED** [review]

**Review by edkaya — APPROVED** [review]

**Review by laurenzfb — APPROVED** [review]

**Review by egekurt123 — APPROVED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**